### ISSUE TITLE
Fix functional test compilation issue and add a PR test

### ISSUE THREAD

**Issue by smlambert** [eclipse-openj9/openj9#14028]
This is a similar situation to https://github.com/eclipse-openj9/openj9/issues/12756 where some test material can not be compiled against a hotspot based build.  Since we use the openj9 functional test suite as part of AQA test suite (in the AQAvit project), we need to ensure test material will compile for any JDK implementation.  

Given that this has happened on several occasions, we should also add a quick litmus test as part of PR testing when test material is added or changed to verify that test material can still compile for a non-OpenJ9 implementation.

**Comment by smlambert** [comment]
Failing to compile for all jdk17+ jobs  
<img width="743" alt="Screen Shot 2021-12-02 at 1 28 49 PM" src="https://user-images.githubusercontent.com/2836948/144481711-e3c9c191-950d-4c63-8b07-8b10dcb8b467.png">

link to one of the failing test jobs:
https://ci.adoptopenjdk.net/job/Test_openjdk17_hs_sanity.functional_aarch64_linux/100/console

```
18:51:42      [javac] /home/jenkins/workspace/Test_openjdk17_hs_sanity.functional_aarch64_linux/aqa-tests/functional/TestUtilities/src/org/openj9/test/attachAPI/TargetManager.java:45: error: package com.ibm.lang.management does not exist
18:51:42      [javac] import com.ibm.lang.management.OperatingSystemMXBean;
```

**Comment by smlambert** [comment]
@renfeiw - we are now trying to compile the TestUtilities folder when running with a hotspot JDK_IMPL, which is failing due to TargetManager having some j9 specific classes.  Can you help remind me where the mechanism is for skipping compilation of that or any other problematic folder?  (I am thinking this changed with the dynamic compilation work, but not sure).

**Comment by renfeiw** [comment]
> @renfeiw - we are now trying to compile the TestUtilities folder when running with a hotspot JDK_IMPL, which is failing due to TargetManager having some j9 specific classes. Can you help remind me where the mechanism is for skipping compilation of that or any other problematic folder? (I am thinking this changed with the dynamic compilation work, but not sure).

Looks like it's related to this recent change: https://github.com/eclipse-openj9/openj9/commit/2c0995169000758c73f5740761760922c436cd93#diff-90e908455556ddb385d32caf135b1413d8b5898d3e311ac5ac3d4badf3344ee0 TestUtilities was added into the compilation process of Java16andUp.

**Comment by llxia** [comment]
Tests in [Java16andUp](https://github.com/eclipse-openj9/openj9/blob/master/test/functional/Java16andUp/playlist.xml) apply to hotspot and [they passed before](https://ci.adoptopenjdk.net/job/Test_openjdk17_hs_sanity.functional_aarch64_linux/94/tapResults/). 
Recently, Java16andUp is updated to include TestUtilities to use its `VersionCheck.major()`. It looks like TestUtilities contains other java code that uses some j9 specific classes. 

I think the solution is to split TestUtilities to openj9 specific utilities and common utilities. And Java16andUp can use the common utilities.

**Comment by smlambert** [comment]
After discussion, we can proceed in an iterative way:
- in the short term, we will exclude this from compilation for hotspot, so we do not fail those test builds - @renfeiw to help
- in medium term, we will look to split TestUtilities into 2 and shift TestManager into the split out part called TestUtilitiesJ9 (which would contain J9 internal classes, while the other part will not) - @smlambert to look
- in long term, we will also look at breaking the JavaXXandUp folders, as if they are large, it makes the targets take longer to run, and may limit our ability to reduce execution time by parallelism

**Comment by smlambert** [comment]
Java15andUp is once again broken for use by non-J9 implementations, failing to compile for hs (https://ci.adoptopenjdk.net/job/Test_openjdk17_hs_sanity.functional_aarch64_linux/111/console) likely due to https://github.com/eclipse-openj9/openj9/pull/14189.

**Comment by pshipton** [comment]
If it's related to https://github.com/eclipse-openj9/openj9/pull/14189 then it's because the TestUtilities were pulled in to get the `VersionCheck.classFile()`, and the TestUtilities must be OpenJ9 specific.

**Comment by smlambert** [comment]
Yes, see https://github.com/eclipse-openj9/openj9/issues/14028#issuecomment-988225623, I will split TestUtilities into 2 parts to make more functional tests available for use by all.
